rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -667,12 +667,22 @@ namespace rsx
 
 			u64 local_vblank_count = 0;
 
+			u64 last_system_time = start_time;
+
 			// TODO: exit condition
 			while (!is_stopped())
 			{
 				// Get current time
 				const u64 current = get_system_time();
 
+				if (current - last_system_time >= 20'000u)
```

Comment:
This is a hack. Why doesn't the proper state propagate here?

Reply:
FTR this is also possible under very slow emulation conditions. 1 frame every several dozen frames is possible when debugging e.g when capturing frames for RSX analysis or replaying captures.
I feel the fix here belongs somewhere else or some other logic. OS sleep is not a concern. Debugger pause should be made visible to threads through their state.

Reply:
If the thread execution is paused neither loops work correctly. The loops in question are suppused to count emulation time, not time passed by operating system hibernation or debugger pause etc.

Reply:
I can increaee the VBLANK diff to a second if it's a concern, in the future savestates will be implemented and they can be used to do proper pausing and unpausing. But it is what it is now.
Although it doesn't address external pause by the operating system and I suspect it is required unless we use possible operating system events so we can use to savestate than load it on wake up.

Reply:
or just drop vblank fix for now.